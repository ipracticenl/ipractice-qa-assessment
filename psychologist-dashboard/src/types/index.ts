// Base types
export interface User {
  id: number;
  name: string;
  type: "psychologist" | "client";
}

// Time range base type
export interface TimeRange {
  id: string;
  from: Date;
  to: Date;
}

// Client DTOs
export interface CreateClientDto {
  name: string;
  initialPsychologistIds: number[];
}

export interface ClientSummaryDto {
  id: number;
  name: string;
}

export interface ClientDetailsDto {
  id: number;
  name: string;
  assignedPsychologists: number[];
  bookedAppointments: Appointment[];
}

// Psychologist DTOs
export interface CreatePsychologistDto {
  name: string;
  initialClients: number[];
}

export interface PsychologistSummaryDto {
  id: number;
  name: string;
}

export interface PsychologistDetailsDto {
  id: number;
  name: string;
  assignedClients: number[];
  availableTimeSlots: AvailableTimeSlot[];
  bookedAppointments: BookedAppointment[];
}

// Booking DTOs
export interface CreateBookingDto {
  psychologistId: number;
  availableTimeSlotId: string;
  comment?: string;
}

export interface ClientDto {
  clientId: number;
}

export interface TimeSlotDto {
  from: Date;
  to: Date;
}

// Model types
export interface AvailableTimeSlot extends TimeRange {
  // Inherits id, from, to from TimeRange
}

export interface Appointment extends TimeRange {
  psychologistId: number;
  comment?: string;
}

export interface BookedAppointment extends TimeRange {
  clientId: number;
}

// Legacy types for backward compatibility
export interface Psychologist extends User {
  type: "psychologist";
  clients: string[];
}

export interface Client extends User {
  type: "client";
  psychologist?: string;
}

export interface Availability {
  id: string;
  psychologistId: number;
  startTime: Date;
  endTime: Date;
  isBooked: boolean;
  clientId?: string;
  clientName?: string;
}
