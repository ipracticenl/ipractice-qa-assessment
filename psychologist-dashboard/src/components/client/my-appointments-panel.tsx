import { Calendar, Clock, UserCheck, MessageSquare } from "lucide-react";
import { format } from "date-fns";
import { Card, CardHeader, CardTitle, CardContent } from "../ui/card";
import {
  Table,
  TableHeader,
  TableRow,
  TableHead,
  TableBody,
  TableCell,
} from "../ui/table";
import { Button } from "../ui/button";
import { Appointment, PsychologistSummaryDto } from "@/types";
import type { FC } from "react";

interface MyAppointmentsPanelProps {
  appointments: Appointment[];
  psychologists: PsychologistSummaryDto[];
  onCancelAppointment: (appointmentId: string) => void;
}

export const MyAppointmentsPanel: FC<MyAppointmentsPanelProps> = ({
  appointments,
  psychologists,
  onCancelAppointment,
}) => {
  return (
    <Card data-testid="my-appointments-panel">
      <CardHeader>
        <CardTitle className="flex items-center gap-2">
          <Calendar className="w-5 h-5" />
          <span className="w-5 h-5 bg-blue-100 rounded-full flex items-center justify-center text-blue-600 text-xs font-bold">
            {appointments.length}
          </span>
          My Appointments
        </CardTitle>
      </CardHeader>
      <CardContent>
        <div className="overflow-x-auto">
          <Table>
            <TableHeader>
              <TableRow>
                <TableHead>Date & Time</TableHead>
                <TableHead>Psychologist</TableHead>
                <TableHead>Comments</TableHead>
                <TableHead>Actions</TableHead>
              </TableRow>
            </TableHeader>
            <TableBody>
              {appointments.length === 0 ? (
                <TableRow>
                  <TableCell
                    colSpan={4}
                    className="text-center py-8 text-gray-500"
                  >
                    No appointments scheduled yet
                  </TableCell>
                </TableRow>
              ) : (
                appointments.map((appointment) => {
                  const psychologist = psychologists.find(
                    (p) => p.id === appointment.psychologistId
                  );
                  return (
                    <TableRow
                      key={appointment.id}
                      data-testid={`appointment-row-${appointment.id}`}
                    >
                      <TableCell className="font-medium">
                        <div className="flex flex-col">
                          <span>{format(appointment.from, "MM/dd/yyyy")}</span>
                          <span className="text-sm text-gray-600 flex items-center gap-1">
                            <Clock className="w-3 h-3" />
                            {format(appointment.from, "HH:mm")} -{" "}
                            {format(appointment.to, "HH:mm")}
                          </span>
                        </div>
                      </TableCell>
                      <TableCell>
                        <div className="flex items-center gap-2">
                          <UserCheck className="w-4 h-4 text-green-600" />
                          {psychologist?.name}
                        </div>
                      </TableCell>
                      <TableCell>
                        {appointment.comment ? (
                          <div className="flex items-start gap-2 max-w-xs">
                            <MessageSquare className="w-4 h-4 text-blue-600 mt-0.5 flex-shrink-0" />
                            <span
                              className="text-sm text-gray-700 break-words"
                              dangerouslySetInnerHTML={{
                                __html: appointment.comment,
                              }}
                            />
                          </div>
                        ) : (
                          <span className="text-sm text-gray-400 italic">
                            No comments
                          </span>
                        )}
                      </TableCell>
                      <TableCell>
                        <Button
                          size="sm"
                          variant="destructive"
                          onClick={() => onCancelAppointment(appointment.id)}
                          className="min-h-[36px] touch-manipulation"
                          data-testid={`cancel-appointment-${appointment.id}`}
                          aria-label={`Cancel appointment with ${
                            psychologist?.name
                          } on ${format(appointment.from, "MM/dd/yyyy HH:mm")}`}
                        >
                          Cancel
                        </Button>
                      </TableCell>
                    </TableRow>
                  );
                })
              )}
            </TableBody>
          </Table>
        </div>
      </CardContent>
    </Card>
  );
};
