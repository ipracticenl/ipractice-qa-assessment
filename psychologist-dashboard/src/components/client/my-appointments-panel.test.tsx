import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { MyAppointmentsPanel } from "./my-appointments-panel";
import { Appointment, PsychologistSummaryDto } from "@/types";

// Mock data with local timezone dates
const mockPsychologists: PsychologistSummaryDto[] = [
  { id: 1, name: "Dr. John Smith" },
  { id: 2, name: "Dr. Jane Doe" },
];

const mockAppointments: Appointment[] = [
  {
    id: "appointment-1",
    psychologistId: 1,
    from: new Date("2024-01-15T10:00:00"),
    to: new Date("2024-01-15T11:00:00"),
  },
  {
    id: "appointment-2",
    psychologistId: 2,
    from: new Date("2024-01-16T14:00:00"),
    to: new Date("2024-01-16T15:00:00"),
  },
];

describe("MyAppointmentsPanel", () => {
  const mockOnCancelAppointment = vi.fn();

  describe("Snapshot Tests", () => {
    it("should match snapshot when no appointments exist", () => {
      // Given: No appointments are provided
      const emptyAppointments: Appointment[] = [];

      // When: The component is rendered with no appointments
      const { container } = render(
        <MyAppointmentsPanel
          appointments={emptyAppointments}
          psychologists={mockPsychologists}
          onCancelAppointment={mockOnCancelAppointment}
        />
      );

      // Then: The component should match the snapshot
      expect(container).toMatchSnapshot();
    });

    it("should match snapshot when appointments exist", () => {
      // Given: Appointments are provided
      const appointments = mockAppointments;

      // When: The component is rendered with appointments
      const { container } = render(
        <MyAppointmentsPanel
          appointments={appointments}
          psychologists={mockPsychologists}
          onCancelAppointment={mockOnCancelAppointment}
        />
      );

      // Then: The component should match the snapshot
      expect(container).toMatchSnapshot();
    });
  });

  describe("Empty State", () => {
    it("should display empty state message when no appointments exist", () => {
      // Given: No appointments are provided
      const emptyAppointments: Appointment[] = [];

      // When: The component is rendered
      render(
        <MyAppointmentsPanel
          appointments={emptyAppointments}
          psychologists={mockPsychologists}
          onCancelAppointment={mockOnCancelAppointment}
        />
      );

      // Then: The empty state message should be displayed
      expect(
        screen.getByText("No appointments scheduled yet")
      ).toBeInTheDocument();
    });

    it("should display correct appointment count in header", () => {
      // Given: No appointments are provided
      const emptyAppointments: Appointment[] = [];

      // When: The component is rendered
      render(
        <MyAppointmentsPanel
          appointments={emptyAppointments}
          psychologists={mockPsychologists}
          onCancelAppointment={mockOnCancelAppointment}
        />
      );

      // Then: The appointment count should be 0
      expect(screen.getByText("0")).toBeInTheDocument();
    });
  });

  describe("Appointments Display", () => {
    it("should display appointments when they exist", () => {
      // Given: Appointments are provided
      const appointments = mockAppointments;

      // When: The component is rendered
      render(
        <MyAppointmentsPanel
          appointments={appointments}
          psychologists={mockPsychologists}
          onCancelAppointment={mockOnCancelAppointment}
        />
      );

      // Then: Appointment rows should be displayed
      expect(
        screen.getByTestId("appointment-row-appointment-1")
      ).toBeInTheDocument();
      expect(
        screen.getByTestId("appointment-row-appointment-2")
      ).toBeInTheDocument();
    });

    it("should display correct appointment count in header", () => {
      // Given: Appointments are provided
      const appointments = mockAppointments;

      // When: The component is rendered
      render(
        <MyAppointmentsPanel
          appointments={appointments}
          psychologists={mockPsychologists}
          onCancelAppointment={mockOnCancelAppointment}
        />
      );

      // Then: The appointment count should be 2
      expect(screen.getByText("2")).toBeInTheDocument();
    });

    it("should display appointment date and time correctly", () => {
      // Given: An appointment is provided
      const appointments = [mockAppointments[0]];

      // When: The component is rendered
      render(
        <MyAppointmentsPanel
          appointments={appointments}
          psychologists={mockPsychologists}
          onCancelAppointment={mockOnCancelAppointment}
        />
      );

      // Then: The date and time should be formatted correctly
      expect(screen.getByText("01/15/2024")).toBeInTheDocument();
      // Check for time range - use getAllByText to handle multiple matches
      // @ts-expect-error - This is a test
      const timeElements = screen.getAllByText((content, element) => {
        return (
          element?.textContent?.includes("10:00") &&
          element?.textContent?.includes("11:00")
        );
      });
      expect(timeElements.length).toBeGreaterThan(0);
    });

    it("should display psychologist name correctly", () => {
      // Given: An appointment with a psychologist is provided
      const appointments = [mockAppointments[0]];

      // When: The component is rendered
      render(
        <MyAppointmentsPanel
          appointments={appointments}
          psychologists={mockPsychologists}
          onCancelAppointment={mockOnCancelAppointment}
        />
      );

      // Then: The psychologist name should be displayed
      expect(screen.getByText("Dr. John Smith")).toBeInTheDocument();
    });

    it('should display "Unknown" when psychologist is not found', () => {
      // Given: An appointment with a non-existent psychologist ID is provided
      const appointmentWithUnknownPsychologist: Appointment = {
        id: "appointment-3",
        psychologistId: 999, // Non-existent ID
        from: new Date("2024-01-15T10:00:00"),
        to: new Date("2024-01-15T11:00:00"),
      };

      // When: The component is rendered
      render(
        <MyAppointmentsPanel
          appointments={[appointmentWithUnknownPsychologist]}
          psychologists={mockPsychologists}
          onCancelAppointment={mockOnCancelAppointment}
        />
      );

      // Then: The psychologist name should be undefined (not displayed)
      const psychologistCell = screen.getByTestId(
        "appointment-row-appointment-3"
      );
      expect(psychologistCell).toBeInTheDocument();
      // The name won't be displayed since it's undefined
    });
  });

  describe("Cancel Appointment Functionality", () => {
    it("should call onCancelAppointment when cancel button is clicked", () => {
      // Given: An appointment is provided and the cancel function is mocked
      const appointments = [mockAppointments[0]];
      const mockCancelFunction = vi.fn();

      // When: The component is rendered and cancel button is clicked
      render(
        <MyAppointmentsPanel
          appointments={appointments}
          psychologists={mockPsychologists}
          onCancelAppointment={mockCancelFunction}
        />
      );

      const cancelButton = screen.getByTestId(
        "cancel-appointment-appointment-1"
      );
      fireEvent.click(cancelButton);

      // Then: The cancel function should be called with the correct appointment ID
      expect(mockCancelFunction).toHaveBeenCalledWith("appointment-1");
    });

    it("should have correct aria-label for cancel button", () => {
      // Given: An appointment is provided
      const appointments = [mockAppointments[0]];

      // When: The component is rendered
      render(
        <MyAppointmentsPanel
          appointments={appointments}
          psychologists={mockPsychologists}
          onCancelAppointment={mockOnCancelAppointment}
        />
      );

      // Then: The cancel button should have the correct aria-label
      const cancelButton = screen.getByTestId(
        "cancel-appointment-appointment-1"
      );
      expect(cancelButton).toHaveAttribute("aria-label");
      expect(cancelButton.getAttribute("aria-label")).toContain(
        "Cancel appointment with Dr. John Smith"
      );
      expect(cancelButton.getAttribute("aria-label")).toContain("01/15/2024");
    });
  });

  describe("Component Structure", () => {
    it("should render the main panel with correct test ID", () => {
      // Given: Appointments are provided
      const appointments = mockAppointments;

      // When: The component is rendered
      render(
        <MyAppointmentsPanel
          appointments={appointments}
          psychologists={mockPsychologists}
          onCancelAppointment={mockOnCancelAppointment}
        />
      );

      // Then: The main panel should be rendered with correct test ID
      expect(screen.getByTestId("my-appointments-panel")).toBeInTheDocument();
    });

    it("should render table headers correctly", () => {
      // Given: Appointments are provided
      const appointments = mockAppointments;

      // When: The component is rendered
      render(
        <MyAppointmentsPanel
          appointments={appointments}
          psychologists={mockPsychologists}
          onCancelAppointment={mockOnCancelAppointment}
        />
      );

      // Then: The table headers should be displayed
      expect(screen.getByText("Date & Time")).toBeInTheDocument();
      expect(screen.getByText("Psychologist")).toBeInTheDocument();
      expect(screen.getByText("Actions")).toBeInTheDocument();
    });
  });
});
