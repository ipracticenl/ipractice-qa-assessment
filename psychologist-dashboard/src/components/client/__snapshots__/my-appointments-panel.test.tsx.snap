// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAppointmentsPanel > Snapshot Tests > should match snapshot when appointments exist 1`] = `
<div>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm"
    data-testid="my-appointments-panel"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <div
        class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-2"
      >
        <svg
          class="lucide lucide-calendar w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 2v4"
          />
          <path
            d="M16 2v4"
          />
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="4"
          />
          <path
            d="M3 10h18"
          />
        </svg>
        <span
          class="w-5 h-5 bg-blue-100 rounded-full flex items-center justify-center text-blue-600 text-xs font-bold"
        >
          2
        </span>
        My Appointments
      </div>
    </div>
    <div
      class="p-6 pt-0"
    >
      <div
        class="overflow-x-auto"
      >
        <div
          class="relative w-full overflow-auto"
        >
          <table
            class="w-full caption-bottom text-sm"
          >
            <thead
              class="[&_tr]:border-b"
            >
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
              >
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Date & Time
                </th>
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Psychologist
                </th>
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Comments
                </th>
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Actions
                </th>
              </tr>
            </thead>
            <tbody
              class="[&_tr:last-child]:border-0"
            >
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                data-testid="appointment-row-appointment-1"
              >
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0 font-medium"
                >
                  <div
                    class="flex flex-col"
                  >
                    <span>
                      01/15/2024
                    </span>
                    <span
                      class="text-sm text-gray-600 flex items-center gap-1"
                    >
                      <svg
                        class="lucide lucide-clock w-3 h-3"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <circle
                          cx="12"
                          cy="12"
                          r="10"
                        />
                        <polyline
                          points="12 6 12 12 16 14"
                        />
                      </svg>
                      10:00
                       -
                       
                      11:00
                    </span>
                  </div>
                </td>
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                >
                  <div
                    class="flex items-center gap-2"
                  >
                    <svg
                      class="lucide lucide-user-check w-4 h-4 text-green-600"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                      />
                      <circle
                        cx="9"
                        cy="7"
                        r="4"
                      />
                      <polyline
                        points="16 11 18 13 22 9"
                      />
                    </svg>
                    Dr. John Smith
                  </div>
                </td>
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                >
                  <span
                    class="text-sm text-gray-400 italic"
                  >
                    No comments
                  </span>
                </td>
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                >
                  <button
                    aria-label="Cancel appointment with Dr. John Smith on 01/15/2024 10:00"
                    class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-destructive text-destructive-foreground hover:bg-destructive/90 h-9 rounded-md px-3 min-h-[36px] touch-manipulation"
                    data-testid="cancel-appointment-appointment-1"
                  >
                    Cancel
                  </button>
                </td>
              </tr>
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
                data-testid="appointment-row-appointment-2"
              >
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0 font-medium"
                >
                  <div
                    class="flex flex-col"
                  >
                    <span>
                      01/16/2024
                    </span>
                    <span
                      class="text-sm text-gray-600 flex items-center gap-1"
                    >
                      <svg
                        class="lucide lucide-clock w-3 h-3"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <circle
                          cx="12"
                          cy="12"
                          r="10"
                        />
                        <polyline
                          points="12 6 12 12 16 14"
                        />
                      </svg>
                      14:00
                       -
                       
                      15:00
                    </span>
                  </div>
                </td>
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                >
                  <div
                    class="flex items-center gap-2"
                  >
                    <svg
                      class="lucide lucide-user-check w-4 h-4 text-green-600"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                      />
                      <circle
                        cx="9"
                        cy="7"
                        r="4"
                      />
                      <polyline
                        points="16 11 18 13 22 9"
                      />
                    </svg>
                    Dr. Jane Doe
                  </div>
                </td>
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                >
                  <span
                    class="text-sm text-gray-400 italic"
                  >
                    No comments
                  </span>
                </td>
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0"
                >
                  <button
                    aria-label="Cancel appointment with Dr. Jane Doe on 01/16/2024 14:00"
                    class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-destructive text-destructive-foreground hover:bg-destructive/90 h-9 rounded-md px-3 min-h-[36px] touch-manipulation"
                    data-testid="cancel-appointment-appointment-2"
                  >
                    Cancel
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MyAppointmentsPanel > Snapshot Tests > should match snapshot when no appointments exist 1`] = `
<div>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm"
    data-testid="my-appointments-panel"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <div
        class="text-2xl font-semibold leading-none tracking-tight flex items-center gap-2"
      >
        <svg
          class="lucide lucide-calendar w-5 h-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 2v4"
          />
          <path
            d="M16 2v4"
          />
          <rect
            height="18"
            rx="2"
            width="18"
            x="3"
            y="4"
          />
          <path
            d="M3 10h18"
          />
        </svg>
        <span
          class="w-5 h-5 bg-blue-100 rounded-full flex items-center justify-center text-blue-600 text-xs font-bold"
        >
          0
        </span>
        My Appointments
      </div>
    </div>
    <div
      class="p-6 pt-0"
    >
      <div
        class="overflow-x-auto"
      >
        <div
          class="relative w-full overflow-auto"
        >
          <table
            class="w-full caption-bottom text-sm"
          >
            <thead
              class="[&_tr]:border-b"
            >
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
              >
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Date & Time
                </th>
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Psychologist
                </th>
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Comments
                </th>
                <th
                  class="h-12 px-4 text-left align-middle font-medium text-muted-foreground [&:has([role=checkbox])]:pr-0"
                >
                  Actions
                </th>
              </tr>
            </thead>
            <tbody
              class="[&_tr:last-child]:border-0"
            >
              <tr
                class="border-b transition-colors hover:bg-muted/50 data-[state=selected]:bg-muted"
              >
                <td
                  class="p-4 align-middle [&:has([role=checkbox])]:pr-0 text-center py-8 text-gray-500"
                  colspan="4"
                >
                  No appointments scheduled yet
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
